### PR TITLE
Fixing problem with edit-feature-overlay button when basemap has changed

### PR DESCRIPTION
**Pull request by xavijam**
Releated: #11474.

There is no new code, basically because it was working ok previously, but when a new baselayer is set, we render the view again in order to decide if events should keep working or not. Good thing is we enable/disable events if the feature is editable or not, so we avoid to check if the proper element has a class, which is a bit unstable. Added a notification when the feature data is requested (It was a TODO so...).

**I made changes over google-maps branch so there is one commit not present in the diff, please take a look [here](https://github.com/CartoDB/cartodb/commit/612be006a1b9596b90cfbcc41ae9e161d4734667). It should be part of this PR**.

**Things to test**
- [ ] When a new base layer (with a different provider) is set, edit a feature button should keep working.
- [ ] Try editing a feature and check a loading notification appears, and when row data is loaded, it should disappear.
- [ ] If user applies a custom query over a layer, edit feature button should be disabled.

### CODE REVIEW

**Comment by xavijam** [comment]
Let's change in this PR, I totally agree, it is a bit weird and probably complex.

**Comment by javierarce** [comment]
@alonsogarciapablo the build is showing some errors in the DI specs. Should we do something about them?

**Comment by alonsogarciapablo** [comment]
@xavijam I suggest you merge `google-maps` branch into this branch (I fixed some tests there yesterday). On the other hand, it looks like failures might actually be related to the EditFeatureOverlay thing... not sure if it has something to do with your changes, but:

```
`TypeError: 'undefined' is not a function (evaluating 'this._editFeatureOverlay.render()') in file:///home/ubuntu/workspace/CartoDB-PR-testing/.grunt/cartodb3-specs.js (line 47636)`
```

**Comment by xavijam** [comment]
I did that, but still failing, let me try again.

**Comment by xavijam** [comment]
They keep failing :(

**Comment by javierarce** [comment]
It looks like now there's just one failing spec: 
 
screen shot 2017-02-13 at 09 46 08
